### ISSUE TITLE
Apply ZClip to multiple modules

### ISSUE THREAD

**Issue by stonezhng** [bluorion-com/ZClip#16]
Hi, thanks for your amazing work on gradient clipping! I try to apply zclip to our training pipelines, but in some cases, we have a separate head besides just CLIP, for example the forward pass looks like:

`out  = self.head(self.clip(input))`

where `self.head` can be just a simple MLP.

If we want to use zclip on both the modules (self.head & self.clip), I make the following attempt:

```python
out = self.head(self.clip(input))
loss = self.loss_fn(out, gt)
loss.backward()

zclip.step(self.clip)
zclip.step(self.head)

self.optim.step()
```

Is this still a valid use of zclip? I know I can just put the additional head inside clip, but still curious if zclip supports multiple modules.

**Comment by zhb2000** [comment]
Calling `zclip.step(...)` twice per iteration will update the EMA statistics twice, leading to incorrect running mean and variance.

I think the following code may fix the issue:

```python
out = self.head(self.clip(input))
loss = self.loss_fn(out, gt)
loss.backward()

zclip.step(torch.nn.ModuleList([self.clip, self.head]))

self.optim.step()
```
